admin.py

```diff
@@ -25,3 +37,55 @@ def decorate(self, *args, **kwargs):
 
   return decorate
 
+
+def RequireAppAndDomainAdmin(oauth_decorator):
+  """Decorator to require the user to be an admin."""
+  def Wrap(func):
+    """Decorator to require the user to be an admin."""
+    def decorate(self, *args, **kwargs):
+      """Actual decorate function that requires admin.
+
+      Args:
+        args: Parameters passed on to the specified function if successful.
+        kwargs: Parameters passed on to the specified function if successful.
+      """
+      user = users.get_current_user()
+      if not user or user is None:
+        logging.error('User is not logged in.')
+        self.abort(403)
+
+      identifier = user.email()
+      if identifier is None or identifier is '':
+        logging.error('No identifier found for the user.')
+        self.abort(403)
+
+      directory_user = None
+      try:
+        directory_service = GoogleDirectoryService(oauth_decorator)
+        directory_user = directory_service.GetUser(identifier)
```

Comment:
It would seem nicer to have the move this logic into the directory service, such as:
directory_service.IsAdminUser(user)